lib/builderator/model.rb

```diff
@@ -0,0 +1,45 @@
+module Builderator
+  module Model
+    ##
+    # Shared model interface
+    ##
+    class Base
+      attr_reader :resources
+
+      def initialize(*args)
+        fetch(*args)
+      end
+
+      def fetch
+        @resources = {}
+      end
+
+      def find(filters = {})
+        Util.filter(resources, filters)
+      end
+
+      def select(set = [])
+        resources.select { |k, _| set.include?(k) }
+      end
+
+      def reject(set)
+        resources.reject { |k, _| set.include?(k) }
+      end
```

Comment:
I'm not too concerned about that TBH. If something fails to pass a filter-set, I'd rather an exception thrown than the reject method fail to filter correctly.